packages/kit/src/runtime/client/client.js

```diff
@@ -375,9 +375,10 @@ async function _goto(url, options, redirect_count, nav_token) {
 
 /** @param {import('./types.js').NavigationIntent} intent */
 async function _preload_data(intent) {
+	const preload_token = (token = {});
```

Comment:
This isn't quite right yet I think. Imagine you navigate to a route, but that's slow, and for some reason there's a preload happening during that time, then the real navigation would be aborted because the preload did update the token. I think we need another global next to token, `preload_token`. It is set inside `_preload_data` and (like now) passed to `load_route` and if the real navigation is a different one to the preloaded one, it resets the `preload_token` so that the if condition in `load_route` doesn't kick in.